awx/ui_next/src/components/ScheduleList/ScheduleList.jsx

```diff
@@ -33,14 +33,17 @@ function ScheduleList({ i18n }) {
   } = useRequest(
     useCallback(async () => {
       const params = parseQueryString(QS_CONFIG, location.search);
+      const response = apiModel
+        ? apiModel.readScheduleList(resource.id, params)
+        : SchedulesAPI.read(params);
```

Comment:
We don't use the `resource` or `apiModel` props for anything but this one network call. I'm wondering if the parent component should wrap the correct api call in a callback and pass it as a prop:

```js
function ScheduleList({ i18n, loadSchedules }) {
   ...
    useCallback(async () => {
      const params = parseQueryString(QS_CONFIG, location.search);
      const response = loadSchedules(params);
```
This would improve the testability of the component:
  - It would remove the conditional assignment, which is effectively another code path to test. 
  - Simpler contract and fewer props overall to test
  - Fewer mocks needed for most tests cases - In some cases the function you pass in could just be a stub promise.